src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs

```diff
@@ -85,6 +100,59 @@ await requestContext.SendResult(new SchemaCompareResult()
             }
         }
 
+        /// <summary>
+        /// Handles request for schema compare generate deploy script
+        /// </summary>
+        /// <returns></returns>
+        public async Task HandleSchemaCompareGenerateScriptRequest(SchemaCompareGenerateScriptParams parameters, RequestContext<ResultStatus> requestContext)
+        {
+            SchemaCompareGenerateScriptOperation operation = null;
+            try
+            {
+                SchemaComparisonResult compareResult = schemaCompareResults.Value[parameters.OperationId];
+                operation = new SchemaCompareGenerateScriptOperation(parameters, compareResult);
+                SqlTask sqlTask = null;
+                TaskMetadata metadata = new TaskMetadata();
+                metadata.TaskOperation = operation;
+                // want to show filepath in task history instead of server and database
+                metadata.ServerName = parameters.ScriptFilePath;
+                metadata.DatabaseName = string.Empty;
+                metadata.Name = "Generate Script";
+
+                sqlTask = SqlTaskManagerInstance.CreateAndRun<SqlTask>(metadata);
+
+                await requestContext.SendResult(new ResultStatus()
+                {
+                    Success = true,
+                    ErrorMessage = operation.ErrorMessage
+                });
+            }
+            catch (Exception e)
+            {
```

Comment:
We are not sending back e.Message. Might miss on some failure message..

Reply:
operation.ErrorMessage gets set to e.Message before the exception gets thrown to this catch